src/geometry/broad_phase_multi_sap/broad_phase_multi_sap.rs

```diff
@@ -90,6 +90,13 @@ pub struct BroadPhaseMultiSap {
     //       Another alternative would be to remove ColliderProxyId and
     //       just use a Coarena. But this seems like it could use too
     //       much memory.
+    #[cfg_attr(
+        feature = "serde-serialize",
+        serde(
+            serialize_with = "parry::utils::hashmap::serialize_hashmap_capacity",
+            deserialize_with = "parry::utils::hashmap::deserialize_hashmap_capacity"
+        )
+    )]
     colliders_proxy_ids: HashMap<ColliderHandle, BroadPhaseProxyIndex>,
```

Comment:
This doesn’t look correct. `serialize_hashmap_capacity` serializes the hashmap as a single value (its capacity) but doesn’t actually serialize its content. So deserialization will result in an emptied `collidiers_proxy_id` which is invalid.

I’m surprised this breaks string serialization though. Is it failing to serialize `ColliderHandle` or `BroadPhaseProxyIndex`?

Reply:
It was failing on the key ; I'll dig more

Reply:
I believe that's the root issue: https://github.com/serde-rs/json/issues/887. I'm surprised there is not a workaround provided in the issue :(

I imagine we could serialize to Vec, and deserialize to HashMap ? But I'm not sure how that would get implemented. And **where**, because we should keep the same hashmap serialization logic for bincode.

Reply:
Mmh, yeah, I’m surprised no alternative was suggested in the serde issue.

> I imagine we could serialize to Vec, and deserialize to HashMap ?

That sounds reasonable. Implementing this would be somewhat similar to `de/serialize_hashmap_capacity`. The serialization functions would look something like that:

```rust
/// Serializes only the capacity of a hash-map instead of its actual content.
#[cfg(feature = "serde-serialize")]
pub fn serialize_hashmap_as_vec<S: serde::Serializer, K, V, H: std::hash::BuildHasher>(
    map: &StdHashMap<K, V, H>,
    s: S,
) -> Result<S::Ok, S::Error> {
    let hashmap_as_vec = /* convert the hashmap to a vec */;
    hashmap_as_vec.serialize(s);
}

/// Creates a new hash-map with its capacity deserialized from `d`.
#[cfg(feature = "serde-serialize")]
pub fn deserialize_hashmap_capacity<
    'de,
    D: serde::Deserializer<'de>,
    K,
    V,
    H: std::hash::BuildHasher + Default,
>(
    d: D,
) -> Result<StdHashMap<K, V, H>, D::Error> {
    let hashmap_as_vec: Vec<(K, V)> = Deserialize::deserialize(deserializer)?;
    /* Convert the hashmap_as_vec to a hashmap */
}

```

> because we should keep the same hashmap serialization logic for bincode

It’s fine if we don’t keep the same logic for bincode.